ucp/_libs/utils.pyx

```diff
@@ -16,23 +18,17 @@ def get_buffer_data(buffer, check_writable=False):
     Returns data pointer of the buffer. Raising ValueError if the buffer
     is read only and check_writable=True is set.
     """
-    iface = None
-    if hasattr(buffer, "__cuda_array_interface__"):
-        iface = buffer.__cuda_array_interface__
-    elif hasattr(buffer, "__array_interface__"):
-        iface = buffer.__array_interface__
+
+    iface = getattr(buffer, "__array_interface__", None)
+    if iface is None:
+        iface = getattr(buffer, "__cuda_array_interface__", None)
```

Comment:
Can we swap these to say the extra `getattr` call in the case of GPU memory? 😄

Reply:
`"__cuda_array_interface__"` is now higher priority.